moveit_ros/visualization/planning_scene_rviz_plugin/src/planning_scene_display.cpp

```diff
@@ -370,8 +370,13 @@ void PlanningSceneDisplay::changedPlanningSceneTopic()
     std::string service_name = planning_scene_monitor::PlanningSceneMonitor::DEFAULT_PLANNING_SCENE_SERVICE;
     if (!getMoveGroupNS().empty())
       service_name = ros::names::append(getMoveGroupNS(), service_name);
-    auto bg_func = [=]() { planning_scene_monitor_->requestPlanningSceneState(service_name); };
-    addBackgroundJob(bg_func, "planning_scene_monitor_->requestPlanningSceneState");
+    auto bg_func = [=]() {
+      if (planning_scene_monitor_->requestPlanningSceneState(service_name))
+        addMainLoopJob(boost::bind(&PlanningSceneDisplay::onNewPlanningSceneState, this));
+      else
+        setStatus(rviz::StatusProperty::Warn, "PlanningScene", "Requesting initial scene failed");
```

Comment:
Does it make sense to retry this later?
I could imagine a roslaunch take more than 5 seconds to bring up a `move_group`, while RViz is up much earlier.
This might happen, e.g., if moveit has to wait for hardware controllers.

Reply:
How often / long do you want to retry? I wouldn't make this too complicated. In case, you can still reset the display.